### PR TITLE
Update Pin Field to guard against setting non numeric values

### LINKED ISSUE #125: [Bug]: Update Pin Field to guard against setting non numeric values
### Components Version

0.5.0

### Which package type are you using?

Unbundled

### What type of bug is this?

Behaviour Error

### Test Case/Reproduction

_No response_

### Description

Currently the Pin Field only has a guard on the keydown event which will limit a user from updating the value of the Pin field via physical or virtual keyboard.

The requirement is to add the guard to the following lit functions

firstUpdated - Invoked when the element is first updated. Implement to perform one time work on the element after update.
attributeChangedCallback - Synchronizes property values when attributes change.

Suggestion is to add a event listener for paste events to ensure that only numeric values can be pasted in the Pin field.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct
- [X] I have [searched for existing issues](https://github.com/capitec/omni-components/issues) and to the best of my knowledge this is not a duplicate

### PR DESCRIPTION
**Pull request by chromaticWaster**
## Description:

closes #125 

- Enhance guard of the Pin Field to stop the setting of non-numeric values.
- Allow the use of keyboard shortcuts for the Pin Field.
- Updates to resolve test failures for stories.

## Test results

image

### Alpha keys
--------------------------------------------------------------------------------------------------------------------------------------------

**Before**

Pin field setting non numeric value

**After**

Pin field setting non numeric value after

### Special characters
--------------------------------------------------------------------------------------------------------------------------------------------
**before (Note period character ".", "+" ,"-" used)**

Pin field setting special characters

**after**

Pin field setting special characters after

### Keyboard shortcut scenarios
-----------------------------------------------------------------

**ctrl+c**

before 

Pin field copy shortcut before

after

Pin field copy shortcut after

--------------------------------------------------------------------------------------------------------------------------------------------

**ctrl+x**

before 

Pin field cut shortcut

after

Pin field cut shortcut after

--------------------------------------------------------------------------------------------------------------------------------------------

**ctrl+v**

before

Pin field paste shortcut

after

Pin field paste shortcut after

### All Submissions:
-----------------------------------------------------------------

* [x] I have followed the [contribution guidelines](https://github.com/capitec/omni-components/blob/develop/CONTRIBUTING.md) for this project.
* [x] I have checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change.

### Changes to Core Features:

* [x] I have added an explanation of what my changes do and why I would like to include them.
* [x] I have written new tests for my core changes, as applicable.
* [x] I have successfully ran tests with my changes locally.
* [x] I have added/updated docs, as applicable.

### CODE REVIEW

File: src/pin-field/PinField.ts [hunk 1]
```diff
@@ -124,8 +139,8 @@ export class PinField extends OmniFormElement {
     }
 
     _keyDown(e: KeyboardEvent) {
-        // Stop alpha keys
-        if (e.key >= 'a' && e.key <= 'z') {
+        // Stop alpha keys unless its a ctrl key combination ie: ctrl+c and specific characters
+        if ((!e.ctrlKey && e.key >= 'a' && e.key <= 'z') || e.key === '-' || e.key === '=' || e.key === '+' || e.key === '.') {
```
Comment (BOTLANNER) [inline src/pin-field/PinField.ts:143]: Does `ctrlKey` cover Mac scenarios?

File: src/pin-field/PinField.ts [hunk 2]
```diff
@@ -142,6 +157,22 @@ export class PinField extends OmniFormElement {
         this.value = input?.value;
     }
 
+    // When a value is pasted in the input.
+    _onPaste(e: ClipboardEvent) {
+        const input = this._inputElement as HTMLInputElement;
+        const clipboardData = e.clipboardData;
+        let pastedData = clipboardData?.getData('Text');
+
+        if (pastedData && new RegExp('^[0-9]+$').test(pastedData) === false) {
```
Comment (BOTLANNER) [inline src/pin-field/PinField.ts:166]: Wont this only run for non numbers?

Comment (chromaticWaster) [reply src/pin-field/PinField.ts:166]: The function will be removed as the attributeChangedCallback function solves paste scenarios

**Comment by chromaticWaster** [comment]
PR will be cancelled. Pin Field implementation will require a updated approach to blocking non-numeric values

**Pull request closed without merging**